src/cpp/session/SessionSourceDatabaseSupervisor.cpp

```diff
@@ -420,10 +420,15 @@ Error removeAndRecreate(const FilePath& dir)
 Error attachToSourceDatabase()
 {  
    // this session may already have a source database; if it does, re-acquire a
-   // lock and then use it
+   // lock and then use it. don't log warnings as this should only fail when
+   // e.g. the filesystem does not support the active locking scheme
    FilePath existingSdb = sessionDirPath();
    if (existingSdb.exists())
-      return sessionDirLock().acquire(sessionLockFilePath(existingSdb));
+   {
+      Error error = sessionDirLock().acquire(sessionLockFilePath(existingSdb));
```

Comment:
We do log a warning if this fails elsewhere. Perhaps we should follow suit here? (or should that place not log, either?)

https://github.com/rstudio/rstudio/blob/a38dbdc6510a8cf007350d4dcd5ccd1afb315147/src/cpp/session/SessionSourceDatabaseSupervisor.cpp#L252-L256

Reply:
Good catch. I think we should log here as well.